RSyntaxTextArea/src/main/java/org/fife/ui/rtextarea/SearchEngine.java

```diff
@@ -72,7 +72,7 @@ public static SearchResult find(JTextArea textArea, SearchContext context) {
 		boolean doMarkAll = textArea instanceof RTextArea && context.getMarkAll();
 
 		String text = context.getSearchFor();
-		if (text==null || text.length()==0) {
+		if (text==null || text.length()==0 || textArea.getText().isEmpty()) {
```

Comment:
This empty-check is expensive for large documents, can you update this to be:

```suggestion
if (text == null || text.isEmpty() || textArea.getDocument().getLength() == 0) {
```